### PR TITLE
Fix broken bash shell after aio app run

### PR DESCRIPTION
**Pull request by moritzraho**
## Description
 fix for https://github.com/adobe/aio-cli-plugin-app/issues/127
 do not process.exit

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by moritzraho** [comment]
@purplecabbage yes the trick was to do bundler.close() to properly clean up the parcel resources, the problem was that parcel had some open file watchers.

**Comment by moritzraho** [comment]
rebased to master to only include relevant commits

**Comment by moritzraho** [comment]
@purplecabbage ready for another review 🙂

**Comment by moritzraho** [comment]
@purplecabbage  Actually went too fast.. I reverted to using process.exit again (reset + force push)
The problems were:
-  aio app run --local wasn't properly cleaning up for some reason
-  could not exit before server is fully running (which is annoying)

I moved the changes for no process.exit to a draft PR: https://github.com/adobe/aio-app-scripts/pull/104
Let's focus on the broken bash shell that is resolved by this PR.

File: test/scripts/dev.test.js [hunk 1]
```diff
@@ -455,23 +473,69 @@ function runCommonWithFrontendTests (ref) {
...
+    // if test times out => means handler is not calling process.exit
+  })
+
+  test('should return another available port for the UI server if used', async () => {
+    mockUIServerAddressInstance.port = 9999
+    const options = { parcel: { https: { cert: 'cert.cert', key: 'key.key' } } }
+    const resultUrl = await ref.scripts.runDev([8888], options)
+    expect(Bundler.mockServe).toHaveBeenCalledWith(8888, options.parcel.https)
+    expect(resultUrl).toBe('https://localhost:9999')
+    expect(mockOnProgress).toHaveBeenCalledWith('Could not use port:8888, using port:9999 instead')
```
Comment (purplecabbage) [inline test/scripts/dev.test.js:529]: nit: Not crazy about tests for specific text output.  This means if we decide to change the text, we break the test, I think the 2 previous expectations are sufficient as they test 'function' and not 'form'

Comment (moritzraho) [reply test/scripts/dev.test.js:529]: yup good point 👍

**Pull request merged**